### ISSUE TITLE
How would you model this system?

### ISSUE THREAD

**Issue by bradduthie** [bradduthie/resevol#46]
Hi All, From some [constructive feedback](https://github.com/bradduthie/resevol/commit/c213c489c84ba1e87beda41dc6cd111a9f182eb5#diff-c5bc2dfcbcf3824db0336fb029e23df4ab2898dfbd396e5a14edac143cea1d39R6) by @MattTinsleyStirling -- I am considering adding a supporting information (SI) document (and package vignette) that provides suggestions for how to model a type of system or situation using the resevol package. That is, if someone said, "How would I model pesticide resistance to GM crops in an annual system?" (or something slightly more specific), the SI would make a suggestion for how to do it. What would be massively helpful here would be if you could:

1. Comment on whether or not you think that this is important for the initial submission to Methods in Ecology and Evolution, or if this would be something to add later to the package.
2. Suggest some systems to model. E.g., can you reply to this asking "How would I model X", where X is a specific system or scenario that you think might be useful to model. 

I do like the idea of having a document that illustrates how these abstract options in the software could usefully be applied to modelling specific systems. It might minimise the risk that readers miss an opportunity to model their system because it's not clear enough how a particular set of options could be applied.

**Comment by bradduthie** [comment]
Note that the possible options for modelling have now increased given Issue #47 because we can now have crops grow (or shrink, I guess) on the landscape between crop rotations. As I type this, I kind of want to now add pesticide deterioration (which would basically be the same code, but for pesticides), but I should probably stop at some point.

**Comment by luc-bussiere** [comment]
Hi @bradduthie and @MattTinsleyStirling: one thought I had after our discussion today about possibly spelling out how to model "simple cases" of genetics of resistance in order to compare with classic models, in response to Brad's comment that there is no dominance. Could we add parameter "k" to a particular locus in order to apply dominance? I get this would be complex if there are many loci, but if just one, maybe it could work? This is the relevant section of the Lynch and Walsh 1998 book that explains how k works: 
<img width="883" alt="Screenshot 2022-12-01 at 21 00 59" src="https://user-images.githubusercontent.com/43408486/205148324-ef561a07-47bc-4b3a-8b67-ff2503efb446.png">

**Comment by luc-bussiere** [comment]
As I kept reading the chapter (which we coincidentally are discussing tomorrow, I stumbled on another relevant passage explaining why dominance is less important for genes of small effect. We could cite this as justification for not implementing a k across many genes, perhaps?

**Comment by bradduthie** [comment]
Agreed @luc-bussiere -- regarding the addition of `k`, it could be added as an option, but this would get tricky sticking it in the software just for one or a small number of loci. I would be tempted (maybe not a bad idea) to have a vector `k` that is of the same length as the genome to specify dominance across the entire genome. This could get tricky though, but it might be nice to have the option, especially for people wanting to make more simple models.

**Comment by bradduthie** [comment]
Ah! You know what a problem is though @luc-bussiere and @MattTinsleyStirling -- dominance is an inherent property of specific alleles (B1 and B2 in the case above), not loci position (which is arbitrary, both in the real world and in the resevol code). Whenever we have a new allele arise through mutation, we would therefore need to figure out its dominance in relation to all of the other alleles. Greta Bocedi experimented with this at one point in IBMs, so I think it is possible, but it might be way more than we can actually do on a reasonable time scale. We would essentially need to have every new allele spawn a complementary `k` (maybe somewhere else in the genome), I think?

**Comment by luc-bussiere** [comment]
Good point @bradduthie  -- my suggestion would only work in the very narrow case where there are only 2 alleles, and no new evolution. I agree implementing this kind of complexity for more alleles is beyond the scope of your short term plans. If someone were super-keen to recreate a simple model, could they for example specify a two locus system, one of which specified the dominance k for the other locus, having only two alleles, for example? Again feel free to ignore if this complexity is getting too difficult. Just thinking about your extra document on how to set up different kinds of models...

**Comment by rosemckeon** [comment]
Hi everyone. to chip in, I think some vignettes demonstrating case use studies would be awesome and very helpful to encourage use of the package. However nice to have that would be, I don't think it's required for a first release.

One approach to take for these could be to decide on the features of the model which we want to showcase then design the scenario around that. For instance, to show off the real world mapping we could take a real field layout and contrast simulations with a proposed new field layout to compare resistance evolution between those two scarios. 

Another approach could be to look at popular systems we can imagine our model being useful to. Maybe those two ideas can be combined.

Would it be crazy to add a line on the README that invites people to email/start an issue if they have a system they're not sure how to get started modeling? That might generate some feedback and ideas for helpful vignettes.
